### PR TITLE
Added checkbox in other settings option to toggle popup confirmation …

### LINKED ISSUE #159: Add checkbox option to toggle on/off confirmation popup before clearing search results.
This is an extension of https://github.com/creativecommons/ccsearch-browser-extension/issues/157. 

The checkbox can be added under the "Other Settings" tab on the options page.  Make sure that by default this feature should be turned off.

### PR DESCRIPTION
**Pull request by techievivek**
Fixes #159 
**Description**
Added checkbox in "other settings" to toggle popup confirmation before clearing the search result.

**Other information**
Bydefault it is not checked i.e. the popup confirmation won't be displayed.

**Checklist:**
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

### CODE REVIEW

**Comment by techievivek** [comment]
@makkoncept Please have a look.

File: src/options/options.html [hunk 1]
```diff
@@ -90,6 +90,11 @@ <h6 class="vocab heading">Store and load the last search results when you open t
             <input type="checkbox" class="vocab choice-field magenta-colored small-sized" id="enable-search-storage-checkbox" />
             <label for="enable-search-storage-checkbox" class="padding-left-smaller">Yes</label><br />
           </div>
+          <h6 class="vocab heading margin-top-medium">Ask for confirmation everytime before clearing search result?</h6>
+          <div class="enable-search-storage-container">
```
Comment (makkoncept) [inline src/options/options.html]: The class should not be `enable-search-storage-container` as this is not the search storage checkbox.

If you look at the `enable-search-storage-container` [rule in _options.scss](https://github.com/creativecommons/ccsearch-browser-extension/blob/master/src/options/sass/components/_options.scss#L17), you will find that it only has `margin-left: 1rem;` property. 

We can do better than this.
- Remove this class from this and the above div and also remove the rule from `_options.scss`. 
- Define a utility class, `margin-left-medium` in [_utilities.scss](https://github.com/creativecommons/ccsearch-browser-extension/blob/master/src/options/sass/base/_utilities.scss) and use that on both the divs.

File: src/options/base.js [hunk 2]
```diff
@@ -8,6 +8,7 @@ const elements = {
   saveFiltersButton: document.getElementById('save-filters'),
   accordionItems: document.querySelectorAll('.accordion .accordion-link'),
   enableSearchStorageCheckbox: document.getElementById('enable-search-storage-checkbox'),
+  enableSearchClearConfirm: document.getElementById('enable-search-clear-confirmation'),
```
Comment (makkoncept) [inline src/options/base.js]: Let's follow the convention here and name the variable `enableSearchClearConfirmCheckbox`.

**Review by makkoncept — CHANGES_REQUESTED** [review]
@techievivek This requires a thorough review. For now, I have added a couple of comments below that you can work on.

File: src/options/base.js (hunk 2 above)
Comment (techievivek) [reply src/options/base.js]: I initially named it `enableSearchClearConfirmCheckbox` but then it is showing linter error.
The line exceed 100 char length.

Comment (makkoncept) [reply src/options/base.js]: Oh, then you can span it to multiple lines. 
like this:
``` javascript
  enableSearchClearConfirm: document.getElementById(
    "enable-search-clear-confirmation",
  ),
```

Comment (techievivek) [reply src/options/base.js]: Ok. Thank you for the help.

**Comment by techievivek** [comment]
@makkoncept Updated the PR please have a look.

**Review by makkoncept — APPROVED** [review]

**Comment by makkoncept** [comment]
@techievivek It's working fine. Please enter the correct issue number in the PR description -> `Fixes #159`. Then, we can merge this.

**Comment by techievivek** [comment]
@makkoncept I am extremely sorry for that silly mistake, I made the same in CC open source repo too.
I have updated it.
Thank you.

**Pull request merged**